File: frontend/src/services/teamService.ts
import API from './api';

export const getAllTeams = async () => {
  const response = await API.get('/teams');
  return response.data;
};

export const createTeam = async (data: any) => {
  const response = await API.post('/teams', data);
  return response.data;
};

export const getMyTeams = async () => {
  const response = await API.get('/teams/my-teams');
  return response.data;
};

export const getTeamMembers = async (teamId: string) => {
  const response = await API.get(`/teams/${teamId}/members`);
  return response.data;
};

export const removeTeamMember = async (teamId: string, userId: string) => {
  const response = await API.delete(`/teams/${teamId}/members/${userId}`);
  return response.data;
};

export const joinTeam = async (inviteCode: string) => {
  const response = await API.post('/teams/join', { inviteCode });
  return response.data;
};

export const deleteTeam = async (teamId: string) => {
  const response = await API.delete(`/teams/${teamId}`);
  return response.data;
};

File: backend/src/main/java/com/esports/service/TeamService.java
package com.esports.service;

import com.esports.dto.TeamDto;
import com.esports.dto.TeamMemberDto;
import java.util.List;
import java.util.UUID;

public interface TeamService {
    TeamDto createTeam(TeamDto teamDto, String userEmail);

    TeamDto joinTeam(String inviteCode, String userEmail);

    List<TeamDto> getAllTeams();

    List<TeamDto> getMyTeams(String userEmail);

    List<TeamMemberDto> getTeamMembers(UUID teamId);

    void removeMember(UUID teamId, UUID userId, String requesterEmail);

    void deleteTeam(UUID teamId, String requesterEmail);
}
